src/FwdState.cc

```diff
@@ -162,6 +163,9 @@ void FwdState::start(Pointer aSelf)
     // just in case; should already be initialized to false
     request->flags.pinned = false;
 
+    // start the clock just before any peer communication
+    request->hier.startPeerClock();
```

Comment:
We should not move this to FwdState::Start() (i.e., to cover URN) because UrnState::start() passes control back to FwdState, constructing an http request based on urn request. I think this place is correct for both cases.